### PR TITLE
IH-113 Add api download template for company

### PR DESCRIPTION
**Pull request by Sotatek-KimMai**

### CODE REVIEW

File: src/main/java/art/heredium/controller/admin/AdminCompanyMembershipController.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import org.springframework.core.io.Resource;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/admin/company-membership")
```
Comment (Sotatek-ThoLe) [inline src/main/java/art/heredium/controller/admin/AdminCompanyMembershipController.java:15]:
```suggestion
@RequestMapping("/api/file/template/company-membership")
```
Đáp mớ này sang FileController.

File: src/main/java/art/heredium/controller/admin/AdminCompanyMembershipController.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+public class AdminCompanyMembershipController {
+
+  @GetMapping("/download-template")
+  public ResponseEntity<Resource> downloadCompanyTemplate() throws IOException {
+    Resource resource = new ClassPathResource("company_template.xlsx");
+
+    HttpHeaders headers = new HttpHeaders();
+    headers.add(HttpHeaders.CONTENT_DISPOSITION, "attachment; filename=company_template.xlsx");
```
Comment (Sotatek-ThoLe) [inline src/main/java/art/heredium/controller/admin/AdminCompanyMembershipController.java:23]:
```suggestion
    headers.add(HttpHeaders.CONTENT_DISPOSITION, "attachment; filename=membership_upload_template.xlsx");
```

File: src/main/java/art/heredium/controller/admin/AdminCompanyMembershipController.java (hunk 1 above)
Comment (Sotatek-KimMai) [reply src/main/java/art/heredium/controller/admin/AdminCompanyMembershipController.java]: I updated in https://github.com/Incuvers-IT/heredium-BE/pull/80/commits/8c4ea2a1e5a773251925faacaa876f7bcbcba37c

File: src/main/java/art/heredium/controller/admin/AdminCompanyMembershipController.java (hunk 2 above)
Comment (Sotatek-KimMai) [reply src/main/java/art/heredium/controller/admin/AdminCompanyMembershipController.java]: I updated in https://github.com/Incuvers-IT/heredium-BE/pull/80/commits/8c4ea2a1e5a773251925faacaa876f7bcbcba37c

File: src/main/java/art/heredium/controller/FileController.java [hunk 3]
```diff
@@ -96,4 +98,21 @@ public ResponseEntity download(
 
     return new ResponseEntity<>(bytes, httpHeaders, HttpStatus.OK);
   }
+
+  @GetMapping("/template/company-membership/download")
+  public ResponseEntity<Resource> downloadCompanyTemplate() throws IOException {
+    Resource resource = new ClassPathResource("company_template.xlsx");
```
Comment (Sotatek-ThoLe) [inline src/main/java/art/heredium/controller/FileController.java:104]:
```suggestion
    Resource resource = new ClassPathResource("company_upload_membership_template.xlsx");
```

File: src/main/java/art/heredium/core/config/security/SecurityConfig.java [hunk 4]
```diff
@@ -59,7 +59,8 @@ public class SecurityConfig {
     "/swagger-ui/**",
     "/v3/api-docs/**",
     "/api/user/posts/enabled-list",
-    "/api/user/posts/{post-id}"
+    "/api/user/posts/{post-id}",
+    "/api/file/template/company-membership/download"
```
Comment (Sotatek-ThoLe) [inline src/main/java/art/heredium/core/config/security/SecurityConfig.java:63]: ko cần thằng này nữa.

Comment (Sotatek-ThoLe) [inline src/main/java/art/heredium/core/config/security/SecurityConfig.java:63]: ko cần thằng này nữa.

Comment (Sotatek-KimMai) [reply src/main/java/art/heredium/core/config/security/SecurityConfig.java]: I updated in https://github.com/Incuvers-IT/heredium-BE/pull/80/commits/2c42771f59b324020dff869441a5278e346cf740

Comment (Sotatek-KimMai) [reply src/main/java/art/heredium/core/config/security/SecurityConfig.java]: I updated in https://github.com/Incuvers-IT/heredium-BE/pull/80/commits/2c42771f59b324020dff869441a5278e346cf740

File: src/main/java/art/heredium/controller/FileController.java (hunk 3 above)
Comment (Sotatek-KimMai) [reply src/main/java/art/heredium/controller/FileController.java]: I updated in https://github.com/Incuvers-IT/heredium-BE/pull/80/commits/2c42771f59b324020dff869441a5278e346cf740

**Review by Sotatek-ThoLe — APPROVED** [review]

**Pull request merged**